webrender/src/texture_cache.rs

```diff
@@ -780,99 +829,105 @@ impl TextureCache {
+                                                                             result.item.requested_rect.origin.x,
+                                                                             result.item.requested_rect.origin.y,
+                                                                             result.item.requested_rect.size.width,
+                                                                             result.item.requested_rect.size.height,
+                                                                             stride,
+                                                                             bpp,
+                                                                             &mut op);
+                        // image itself
+                        op.update(result.item.requested_rect.origin.x, result.item.requested_rect.origin.y,
+                                  result.item.requested_rect.size.width, result.item.requested_rect.size.height,
+                                  bytes, stride);
+                    }
+                    AllocationKind::Standalone => {
+                        let update_op = TextureUpdate {
+                            id: result.item.texture_id,
+                            op: TextureUpdateOp::Create(width,
+                                                        height,
+                                                        format,
+                                                        filter,
+                                                        RenderTargetMode::None,
+                                                        Some(bytes))
+                        };
+
+                        self.pending_updates.push(update_op);
+                    }
                 };
-
-                self.pending_updates.push(border_update_op_top);
-                self.pending_updates.push(border_update_op_bottom);
-                self.pending_updates.push(border_update_op_left);
-                self.pending_updates.push(border_update_op_right);
-
-                TextureUpdateOp::Update(result.item.requested_rect.origin.x,
-                                        result.item.requested_rect.origin.y,
-                                        width,
-                                        height,
-                                        bytes,
-                                        stride)
             }
-            AllocationKind::Standalone => {
-                TextureUpdateOp::Create(width,
-                                        height,
-                                        format,
-                                        filter,
-                                        RenderTargetMode::None,
-                                        Some(bytes))
+            ImageData::ExternalBuffer(id) => {
+                match result.kind {
+                    AllocationKind::TexturePage => {
+                        let update_op = TextureUpdate {
```

Comment:
Please check whether if it's still a problem after reordering the match expressions in my new patch.